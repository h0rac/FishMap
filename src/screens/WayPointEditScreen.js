import React from 'react';
import { Text, View, StyleSheet,TouchableOpacity, TextInput, Platform } from 'react-native';
import Modal from 'react-native-modal'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {EditedItem} from '../../src/components/EditedItem'
import { Button } from 'react-native-elements';

class WayPointEditScreen extends React.Component {

  static navigationOptions = {
    headerStyle: {
      backgroundColor: '#2F95D6'
    },
    title: 'Waypoint Editor',
    headerTintColor: 'white'
  };

  constructor() {
    super()
    this.state = {
      modalVisible: false,
      ios: false,
      inputValue: null
    }
  }

  componentDidMount() {
		if (Platform.OS === 'ios') {
			this.setState({ ios: true });
    }
  }

  setModalVisible = (status) => {
    this.setState({modalVisible: status})
  }

  handleEdit = (data) => {
    this.setState({modalVisible: !this.state.modalVisible, inputValue: data})
  }

render() {
	return (
    <View style={styles.container}>
    <View style={styles.section}>
    <Text style={styles.sectionText}>Waypoint data</Text>
    </View>
      <EditedItem selectedPosition={this.props.selectedPosition} onEdit={this.handleEdit.bind(this)}/>
      <Modal isVisible={this.state.modalVisible}>
          <View style={{ flex: 1, justifyContent:'center', flexDirection:'column'}}>
            <View style={{flex: 1, backgroundColor: 'white', justifyContent:'center', borderWidth:1, borderRadius:15}}>
            <View style={{flex:0.2, flexDirection:'column', justifyContent:'center', alignItems:'center', borderBottomWidth:1, marginBottom:16}}>
              <Text>Here you can edit your fishmark data and save</Text>
            </View>
            <TextInput style={!this.state.ios ? styles.input : styles.inputIOS}
              onChangeText={(email) => this.setState({ email: email })}
							value={this.state.inputValue}
							placeholderTextColor={'lightgray'}
							underlineColorAndroid='#2F95D6'
							maxLength={40}/>
            <View style={{flex:1, flexDirection:'row', justifyContent:'center', alignItems:'center'}}>
              <Button
              rounded={true}
              icon={{name:"check", type:'font-awesome'}}
              backgroundColor="green"
              onPress={() => this.setState({modalVisible: false})}
              title={'Confirm'}
            >
            </Button>
            <Button
              rounded={true}
              icon={{name:"times", type:'font-awesome'}}
              backgroundColor="red"
              onPress={() => this.setState({modalVisible: false})}
              title={'Cancel'}
            >
            </Button>
          </View>
            </View>
          </View>
        </Modal>
    </View>
    
    )
  }
}
WayPointEditScreen.propTypes = {
  selectedPosition: PropTypes.object
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'whitesmoke',
    flexDirection: 'column'
  },
  input: {
		color: 'gray'

	},

	inputIOS: {
		color: 'gray',
		borderBottomColor: '#2F95D6', // for IOS
		borderBottomWidth: 1, //for IOS
    height: 50,
    marginLeft:10, 
    marginRight:10,
	},
  section: {
    backgroundColor: 'whitesmoke',
    flex: 0.05,
    paddingLeft: 10,
    paddingTop: 30,
    flexDirection: 'row',
    alignItems: 'center'
  },
  
  
  sectionText: {
    color: 'gray'
  },
  });

  const mapStateToProps = state => 
  {
    return {
      selectedPosition: state.fishmarks.selectedPosition
    }
  }

export default connect(mapStateToProps)(WayPointEditScreen);
