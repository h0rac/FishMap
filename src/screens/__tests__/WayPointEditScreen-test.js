import 'react-native';
import React from 'react';
import WayPointEditScreen from '../WayPointEditScreen';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('Testing WayPointEditScreen screen', () => {

	let tree;
	let navigate;
	let state;
	let props;
	let dispatch;

	const event = {
		nativeEvent: {
			coordinate: {
				latitude: 54.475408,
				longitude: 18.263086,
				latitudeDelta: 0.0922,
				longitudeDelta: 0.0421
			}
		}
	};

	beforeEach(() => {
		navigate = jest.fn();
		state = {
			fishmarks: { fishmarks: [] },
			user: { position: {}, duration: 1000 },
			selectedPosition: {latitude: 54.475408,
				longitude: 18.263086,
				latitudeDelta: 0.0922,
				longitudeDelta: 0.0421}

		};
		const store = mockStore(state);
		dispatch = jest.fn();

		props = store;

		tree = shallow(
			<WayPointEditScreen {...props} store={store} navigation={{ navigate }}
			/>, { lifecycleExperimental: true });
	});
	it('renders correctly', () => {
		expect(tree).toMatchSnapshot();
	});

	it('should has WaypointEditList', () => {
		
	})
})