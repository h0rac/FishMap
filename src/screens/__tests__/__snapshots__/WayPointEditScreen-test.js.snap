// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing WayPointEditScreen screen renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(WayPointEditScreen)
    clearActions={[Function]}
    dispatch={[Function]}
    getActions={[Function]}
    getState={[Function]}
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    replaceReducer={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    subscribe={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "navigation": Object {
        "navigate": [MockFunction],
      },
      "replaceReducer": [Function],
      "selectedPosition": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "subscribe": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "navigation": Object {
          "navigate": [MockFunction],
        },
        "replaceReducer": [Function],
        "selectedPosition": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "subscribe": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
    "lifecycleExperimental": true,
  },
}
`;
